### PR TITLE
Update type-specific docs

### PR DESCRIPTION
**Pull request by therealsilence**
- Update loc and syn:tag sections to correct / clarify behavior with prefixes, wildcards, etc.
- Add note to syn:tag to cross-reference tag glob docs when lifting/filtering on tagged nodes
- Add sections for duration and taxonomy types
- Other minor updates / clarifications.

### CODE REVIEW

**Review by invisig0th — DISMISSED** [review]

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 1]
```diff
@@ -1099,14 +1197,29 @@ Lift all the DNS A nodes whose observation window overlaps with the interval of
 ``ival`` types cannot be used with the ``*range=`` custom comparator. ``*range=`` can only be used to specify a
 range of individual values (such as ``time`` or ``int``).
 
+.. TIP::
+  
+  By definition, an ``ival`` is a **pair** of date/time values treated as a single cobmined value. This means that
+  when using variables_ to work with ``ival`` types, the full value of the property is assigned to the variable by
+  default. For example, given a node (such as an ``inet:dns:a`` node) with a ``.seen`` value of ``(2023/07/08 11:19:02, 2023/12/14 21:18:47.612)``,
```
[the remark this answers is not in the document]
Comment (therealsilence) [reply docs/synapse/userguides/storm_ref_type_specific.rstorm:1204]: Copypasta'ed the timestamps from an arbitrary record in Synapse. Will adjust to limit confusion.

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 2]
```diff
@@ -215,6 +223,96 @@ nodes where the ``ou:name`` is present in the ``:name`` property or as an elemen
...
+23 seconds:
+
+.. storm-cli:: [ risk:compromise = * :name = 'example compromise' :duration = '6D 06:46:23' ]
+
+Or:
+
+.. storm-cli:: [ risk:compromise = * :name = 'example compromise' :duration = (542783000) ]
+
+Values can be set at any level of granularity; e.g., ``[ :duration = 19D ]`` is acceptable, as are
+values to milisecond resolution.
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_type_specific.rstorm:278]:
```suggestion
values to millisecond resolution.
```

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 3]
```diff
@@ -215,6 +223,96 @@ nodes where the ``ou:name`` is present in the ``:name`` property or as an elemen
...
+
+duration
+--------
+
+``duration`` is a type that represents a period (length) of time.
+
+Indexing
+++++++++
+
+A ``duration`` is stored as an integer value representing the number of epoch milliseconds (epoch millis).
```
[the remark this answers is not in the document]
Comment (therealsilence) [reply docs/synapse/userguides/storm_ref_type_specific.rstorm]: Removed 'epoch'

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 4]
```diff
@@ -215,6 +223,96 @@ nodes where the ``ou:name`` is present in the ``:name`` property or as an elemen
...
+  
+  While start, end, and duration values are related, Synapse does **not** currently calculate values
+  for these properties. That is, entering a start time and a duration will not automatically calculate
+  and set the ending time. Similarly, if all three values are set and you modify one of the values,
+  the other value(s) are not updated to reflect the change (e.g., changing the duration will not
+  automatically update the ending time).
+  
+  That said, in cases where a node has start, end, and duration properties and you have two of the
+  three values, you can use variables_ to calculate and set the third value. For example, given a
+  ``risk:compromise:name = 'my compromise``  with ``:time = '2023/09/22 10:40'`` and
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_type_specific.rstorm:296]:
```suggestion
  ``risk:compromise:name = 'my compromise'``  with ``:time = '2023/09/22 10:40'`` and
```

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 5]
```diff
@@ -215,6 +223,96 @@ nodes where the ``ou:name`` is present in the ``:name`` property or as an elemen
...
+  
+  ``risk:compromise:name='my compromise' $min=:time $max=:lasttime [ :duration=( $max - $min ) ]``
+  
+  This will set ``:duration = 14D 03:20:00.000``.
+
+
+Operations
+++++++++++
+
+As ``duration`` types are stored as integers and support any operations suitable for those types. This
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_type_specific.rstorm:307]: Should be "they support any operations suitable"?

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 6]
```diff
@@ -1262,123 +1401,251 @@ Strings and string-derived types can also be lifted or filtered using the regula
...
+  on working with tags applied to nodes.
+
+
+.. _type-taxoxnomy:
+
+taxonomy
+--------
+
+A ``taxonomy`` in Synapse is used to represent a hierarchical set of categories that can be used to classify
+objects (forms). Many forms in Synapse include ``taxonomy`` properties (such as ``ou:camptype`` or ``risk:tool:software:taxonomy``).
```
[the remark this answers is not in the document]
Comment (therealsilence) [reply docs/synapse/userguides/storm_ref_type_specific.rstorm]: Updated

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 7]
```diff
@@ -215,6 +223,96 @@ nodes where the ``ou:name`` is present in the ``:name`` property or as an elemen
...
+A ``duration`` is commonly specified using a string value (days / hours / minutes / seconds / milliseconds
+as appropriate) with the following notation:
+  
+  ``##D hh:mm:ss.mmm``
+
+The literal uppercase letter "D" is used to represent the number of days. When entering a duration value as a
+string, single or double quotes are required in accordance with the standard rules for using
+:ref:`storm-whitespace-literals`.
+  
+A ``duration`` can also be specified as the number of epoch milliseconds, enclosed in parentheses:
```
[the remark this answers is not in the document]
Comment (therealsilence) [reply docs/synapse/userguides/storm_ref_type_specific.rstorm]: Tweaked a bit - lmk if this addresses your comment

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 8]
```diff
@@ -215,6 +223,96 @@ nodes where the ``ou:name`` is present in the ``:name`` property or as an elemen
...
+Operations
+++++++++++
+
+As ``duration`` types are stored as integers and support any operations suitable for those types. This
+includes:
+
+- comparison using mathematical operators (such as greater than ( ``>``));
+- finding low or high values using the :ref:`storm-min` or :ref:`storm-max` commands;
+- use of the range ( ``*range=`` ) comparison operator;
+- etc.
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_type_specific.rstorm:313]: `etc.` seems to be formatted as an item in the list, which looks different from how it is used in other lists.

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 9]
```diff
@@ -1099,14 +1197,29 @@ Lift all the DNS A nodes whose observation window overlaps with the interval of
 ``ival`` types cannot be used with the ``*range=`` custom comparator. ``*range=`` can only be used to specify a
 range of individual values (such as ``time`` or ``int``).
 
+.. TIP::
+  
+  By definition, an ``ival`` is a **pair** of date/time values treated as a single cobmined value. This means that
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_type_specific.rstorm:1202]:
```suggestion
  By definition, an ``ival`` is a **pair** of date/time values treated as a single combined value. This means that
```
I do like `cobmined` though

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 10]
```diff
@@ -1262,123 +1401,251 @@ Strings and string-derived types can also be lifted or filtered using the regula
...
 
-- ``syn:tag = rep.vt.exploit``
-- ``syn:tag = aka.kaspersky.mal.shamoon.2``
-- ``syn:tag = cno.tgt.cn_mil_pla``
+.. storm-cli:: [ syn:tag = rep.us-cisa.LAPSUS$ ]
 
-The following ``syn:tag`` values are not allowed and will generate ``BadTypeValu`` errors:
+In the above example:
+
+- the dash in ``us-cisa`` is convereted to an underscore;
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_type_specific.rstorm:1449]:
```suggestion
- the dash in ``us-cisa`` is converted to an underscore;
```

File: docs/synapse/userguides/storm_ref_type_specific.rstorm [hunk 11]
```diff
@@ -1262,123 +1401,251 @@ Strings and string-derived types can also be lifted or filtered using the regula
...
+  
+  Some Synapse `Power-Ups`_ that create forms with ``taxonomy`` properties may set the property value to
+  identify the **source** of the form itself, along with any relevant category provided by the source.
+  This is done in order to distinguish categorizable objects from third-parties from similar objects
+  that may be defined by your organization, in case your internal categories differ from those used by
+  others.
+  
+  For example, the Synapse-Mandiant commercial Power-Up may create industries (``ou:industry`` nodes)
+  whose ``:type`` value will be set to ``mandiant``. Similarly, Mandiant categorizes threat actor tools
+  and malware, so ``risk:tool:software`` nodes created by the Power-Up will have their ``:taxonomy``
```
Comment (Cisphyx) [inline docs/synapse/userguides/storm_ref_type_specific.rstorm:1565]:
```suggestion
  and malware, so ``risk:tool:software`` nodes created by the Power-Up will have their ``:type``
```

**Review by Cisphyx — APPROVED** [review]

**Review by invisig0th — APPROVED** [review]

**Pull request merged**